vmm/src/sigsys_handler.rs

```diff
@@ -15,37 +20,105 @@ use logger::{Metric, METRICS};
 // See https://github.com/rust-lang/libc/issues/716 for why the offset is different in Rust.
 const SI_OFF_SYSCALL: isize = 6;
 
+const SYS_SECCOMP_CODE: i32 = 1;
+
+// This no longer relies on sys_util::register_signal_handler(), which is a lot weirder than it
+// should be (at least for this use case). Also, we want to change the sa_mask field of the
+// sigaction struct.
 /// Sets up the specified signal handler for `SIGSYS`.
-///
-pub fn setup_sigsys_handler() -> Result<(), sys_util::Error> {
-    return unsafe {
-        sys_util::register_signal_handler(
-            libc::SIGSYS,
-            sys_util::SignalHandler::Siginfo(sigsys_handler),
-            false,
-        )
-    };
+pub fn setup_sigsys_handler() -> Result<(), io::Error> {
```

Comment:
Since this function now returns meaningful errors we should also change the expect from src/main.rs and print the error as well. You can use to_string() in io::Error to pretty print it.